In [1]:
from sqlalchemy import create_engine
from sqlalchemy_utils import database_exists, create_database
import pandas as pd

In [2]:
cnx = create_engine("postgresql://ubuntu:ubuntu@localhost:5432/NY_restaurant")

In [3]:
print(pd.read_sql_query('''select * from nyc_inspections order by grade_date asc LIMIT 5 ''',cnx))

      camis                  dba       boro building               street  \
0  40962072        RARYTAS MANOR     QUEENS     5509  METROPOLITAN AVENUE   
1  41509586                BONGO  MANHATTAN      395          WEST STREET   
2  41509586                BONGO  MANHATTAN      395          WEST STREET   
3  41509586                BONGO  MANHATTAN      395          WEST STREET   
4  41509686  EL TEQUILERO II BAR   BROOKLYN     1915  CONEY ISLAND AVENUE   

  zipcode       phone cuisine_description inspection_date  \
0   11385  7183267511    Eastern European      2012-06-23   
1   10014  2126756555            American      2013-04-26   
2   10014  2126756555            American      2013-04-26   
3   10014  2126756555            American      2013-04-26   
4   11230  3474629090             Mexican      2013-05-09   

                                            action violation_code  \
0  Violations were cited in the following area(s).            10H   
1  Violations were cited in the 

The data stretches back to mid 2012, at least when looking at grade_date. 

In [4]:
print(pd.read_sql_query('''select * from nyc_inspections order by inspection_date asc LIMIT 50 ''',cnx))

       camis                      dba       boro building  \
0   50075055                     None  MANHATTAN      555   
1   50075668                 NEW ASIA  MANHATTAN      666   
2   50071835                 ZVAH INC  MANHATTAN       37   
3   50074316              LOU DE FONG   BROOKLYN     6916   
4   50076557                 O2 K-BBQ     QUEENS     4553   
5   50077375         HUNAN SLURP SHOP  MANHATTAN      112   
6   50073006                 STAX NYC  MANHATTAN     1485   
7   50072902                     None     QUEENS     2235   
8   50077397                     None     QUEENS    14720   
9   50073333            UPTOWN MARKET     QUEENS        1   
10  50077488                     None   BROOKLYN     1901   
11  50076070                     None  MANHATTAN       33   
12  50075962           CHAMPION PIZZA  MANHATTAN       17   
13  50074275                     None  MANHATTAN      235   
14  50075296     ABOUT TIME SOUL FOOD     QUEENS     8902   
15  50069905         NUM

In [5]:
print(pd.read_sql_query('''select count(*) from nyc_inspections where inspection_date = '1900-01-01' ''',cnx))

   count
0   1135


### So there are 1135 records that have bad inspection_date fields. It looks like these are bad rows and have zero relevant information, so I'll probably drop these.

In [6]:
print(pd.read_sql_query('''select * from nyc_inspections where inspection_date = '1900-01-01' ''',cnx))

         camis                                     dba           boro  \
0     50045489                        ELMA'S OF HARLEM      MANHATTAN   
1     50073770                              PITA POINT       BROOKLYN   
2     50076692                      PEPPERCORN KITCHEN      MANHATTAN   
3     50062654                      BUFFALO JO'S WINGS         QUEENS   
4     50072938                               ZONI CAFE       BROOKLYN   
5     50075740                                    None       BROOKLYN   
6     50075623                    MALYUTKA BAKERY CAFE       BROOKLYN   
7     50069936                                    None       BROOKLYN   
8     50076117                                    None         QUEENS   
9     50077154                                    None       BROOKLYN   
10    50076397                                 GROUP M      MANHATTAN   
11    50073097                       ROWE'S RESTAURANT       BROOKLYN   
12    50077411                                    N

### Let's find some other "bad" data

In [7]:
print(pd.read_sql_query('''select distinct(grade), count(grade) from nyc_inspections where 
inspection_date != '1900-01-01'
group by grade
order by count(grade) desc
''',cnx))


            grade   count
0               A  148556
1               B   23692
2               C    6106
3               Z    3849
4  Not Yet Graded    2192
5               P    1769
6               G       3
7            None       0


* N = Not Yet Graded
* A = Grade A
* B = Grade B
* C = Grade C
* Z = Grade Pending
* P= Grade Pending issued on re-opening following an initial inspection that resulted in a closure 

In [8]:
print(pd.read_sql_query('''select * from nyc_inspections where inspection_date != '1900-01-01' LIMIT 5''',cnx))

      camis                       dba       boro building           street  \
0  41234254                      TREE  MANHATTAN      190         1 AVENUE   
1  50000738  D'ANNA'S DELI & PIZZERIA     QUEENS    16908    CROCHERON AVE   
2  41421237            BILLY'S BAKERY  MANHATTAN       75  FRANKLIN STREET   
3  50011937        MUSCLE MAKER GRILL  MANHATTAN     1099          2ND AVE   
4  41069367              PAPHOS DINER   BROOKLYN     2501    FULTON STREET   

  zipcode       phone                                cuisine_description  \
0   10009  2123587171                                             French   
1   11358  7188780042  Latin (Cuban, Dominican, Puerto Rican, South &...   
2   10013  2126479958                                             Bakery   
3   10022  6468959533                                           American   
4   11207  7183462301                                           American   

  inspection_date                                           action  \
0   

In [12]:
print(pd.read_sql_query('''select count(*) from nyc_inspections where inspection_date != '1900-01-01' 
and grade is NULL 

LIMIT 5

''',cnx))

    count
0  185003


In [13]:
print(pd.read_sql_query('''select count(*) from nyc_inspections where grade is NULL 

LIMIT 5

''',cnx))

    count
0  186137


### I'll have to drop these rows as well - we wont need any rows without grades

In [15]:
print(pd.read_sql_query('''select count(*) from nyc_inspections where grade is not NULL 
and
inspection_date != '1900-01-01' 

LIMIT 5

''',cnx))

    count
0  186167


### Looks like we'll have 186K rows to play with

In [16]:
print(pd.read_sql_query('''select * from nyc_inspections where grade is not NULL 
and
inspection_date != '1900-01-01' 

LIMIT 5

''',cnx))

      camis                                     dba       boro building  \
0  50011937                      MUSCLE MAKER GRILL  MANHATTAN     1099   
1  41350559  BURGER KING, POPEYES LOUISIANA KITCHEN      BRONX       60   
2  50065144                                DR SMOOD  MANHATTAN      470   
3  50032737                           KAM'S KITCHEN     QUEENS    21709   
4  41569301               KING SUNSHINE JERK CENTER      BRONX     1853   

               street zipcode       phone cuisine_description inspection_date  \
0             2ND AVE   10022  6468959533            American      2017-04-03   
1   METROPOLITAN OVAL   10462  7188223678          Hamburgers      2015-05-20   
2           BROOME ST   10013  3473108262    Café/Coffee/Tea      2017-08-29   
3         LINDEN BLVD   11411  7185251167             Chinese      2016-09-14   
4  WESTCHESTER AVENUE   10472  7184093809           Caribbean      2015-06-04   

                                            action violation_c

In [17]:
print(pd.read_sql_query('''select camis,dba,grade_date,grade from nyc_inspections where grade is not NULL 
and
inspection_date != '1900-01-01' 
group by camis,dba,grade_date,grade

LIMIT 5

''',cnx))

      camis                             dba  grade_date grade
0  50039396       SUGAR FACTORY MEATPACKING  2018-01-25     A
1  40592566                          FRED'S  2015-04-15     A
2  41560744  JOHN'S PIZZERIA AND RESTAURANT  2018-01-19     A
3  50038000                 HOP WON EXPRESS  2015-08-26     C
4  50000061              XI'AN FAMOUS FOODS  2017-04-04     A


In [23]:
print(pd.read_sql_query('''select camis,dba,grade_date,grade from nyc_inspections where grade is not NULL 
and
inspection_date != '1900-01-01' 
group by camis,dba,grade_date,grade
''',cnx))

          camis                               dba  grade_date grade
0      50039396         SUGAR FACTORY MEATPACKING  2018-01-25     A
1      40592566                            FRED'S  2015-04-15     A
2      41560744    JOHN'S PIZZERIA AND RESTAURANT  2018-01-19     A
3      50038000                   HOP WON EXPRESS  2015-08-26     C
4      50000061                XI'AN FAMOUS FOODS  2017-04-04     A
5      41629116                  Asian Kitchen 55  2017-08-21     A
6      50001340             BLUESTONE LANE COFFEE  2016-07-11     A
7      50039131                     PRET A MANGER  2018-04-25     A
8      40902021               BEACH STREET EATERY  2015-04-10     A
9      40899816                  ANDY'S CAFETERIA  2015-03-30     A
10     41337035              VALENCIA COFFEE ROOM  2017-01-19     A
11     41666448  GUARDIAN LIFE EMPLOYEE CAFETERIA  2016-09-07     A
12     50033458                   UPHOLSTRY STORE  2015-06-10     A
13     50072310                SASHIMI EXPRESS I

### 81,031 rows of restaurant, date, grade

In [31]:
print(pd.read_sql_query('''select camis,dba,count(camis) from nyc_inspections where grade is not NULL 
and
inspection_date != '1900-01-01' 
group by camis,dba
having count(camis)>1
order by count(camis) desc
''',cnx))

#GANG SAN DEUL

          camis                                      dba  count
0      50001880                            GANG SAN DEUL     40
1      41630632                               J J NOODLE     35
2      41683816                  MAX BAKERY & RESTAURANT     35
3      40790556              BOSTON JERK-CITY RESTAURANT     34
4      50035784                      BIG WONG RESTAURANT     33
5      50033122  PARTY WELL RESTAURANT & ORIENTAL BAKERY     30
6      50001693                               ANKA GRILL     30
7      50003580                            XING WONG BBQ     30
8      50018922                                 YO SUSHI     29
9      40645073                             LA SABROSURA     29
10     50001637                       YOLANDA RESTAURANT     29
11     41692971                         B BO SING BAKERY     29
12     41403222                 YOLIE'S BAR & RESTAURANT     28
13     41503140                               VIP COFFEE     28
14     41669112            LA VIE EN SZE

In [48]:
print(pd.read_sql_query('''select camis,dba,grade_date,
grade,boro,zipcode,cuisine_description ,inspection_type,building,street

from nyc_inspections 
where grade is not NULL 
and
inspection_date != '1900-01-01'

group by camis,dba,grade_date,
grade,boro,zipcode,cuisine_description ,inspection_type,building,street
''',cnx))

#

          camis                                                dba  \
0      40396231                                  DELMONICO GOURMET   
1      50045568                               STEINWAY COFFEE SHOP   
2      50045092                                 HOT DOG CONCESSION   
3      41491314                     MARIA COFFEE SHOP & RESTAURANT   
4      40576260                              HECTOR'S CAFE & DINER   
5      50046735                                             SUBWAY   
6      41615280                               FLORIDITA RESTAURANT   
7      50040359            SANU RIA MALAYSIAN & INDONESIAN CUISINE   
8      41125267                                    EL PILON BAKERY   
9      50012730                           CAMPO BELLO RESTAURANT 2   
10     40589546                               EL CHIVITO D'ORO III   
11     41337501                         MANDATO MEXICAN RESTAURANT   
12     50018936                            ZOU'S RESTAURANT BUFFET   
13     50007040     

### So we have approximately 80,400 distinct grades given out

In [ ]:
print(pd.read_sql_query('''
SELECT
camis
,dba
,grade_date
,grade,boro
,zipcode
,cuisine_description
,inspection_type
,building
,street
--,*
FROM nyc_inspections 
WHERE grade IS NOT NULL 
AND grade_date IS NOT NULL
AND inspection_date != '1900-01-01' 
AND dba = 'ELUDZ LOUNGE'
GROUP BY camis,dba,grade_date,grade,boro,zipcode,cuisine_description,inspection_type,building,street
ORDER BY grade_date desc
LIMIT 50
''',cnx))

#

In [81]:
print(pd.read_sql_query('''
SELECT boro, grade, count(boro)
FROM nyc_inspections 
WHERE grade IS NOT NULL 
AND grade_date IS NOT NULL
AND inspection_date != '1900-01-01' 
GROUP BY boro,grade
order by 1

''',cnx))

             boro           grade  count
0           BRONX               Z    238
1           BRONX               P    175
2           BRONX               B   2457
3           BRONX               A  13361
4           BRONX               C    652
5        BROOKLYN  Not Yet Graded      3
6        BROOKLYN               C   1545
7        BROOKLYN               B   5996
8        BROOKLYN               Z   1138
9        BROOKLYN               P    541
10       BROOKLYN               A  36289
11      MANHATTAN               G      3
12      MANHATTAN               C   2408
13      MANHATTAN               Z   1465
14      MANHATTAN               A  59554
15      MANHATTAN               P    623
16      MANHATTAN               B   9019
17         QUEENS  Not Yet Graded      2
18         QUEENS               Z    902
19         QUEENS               B   5403
20         QUEENS               C   1321
21         QUEENS               P    371
22         QUEENS               A  34252
23  STATEN ISLAN